pkg/network/mycelium/config.go

```diff
@@ -0,0 +1,56 @@
+package mycelium
+
+import (
+	"context"
+	"crypto/ed25519"
+	"encoding/hex"
+
+	"github.com/decred/base58"
+	"github.com/pkg/errors"
+)
+
+type NodeConfig struct {
+	KeyFile    string
+	TunName    string
+	Peers      []string
+	PrivateKey ed25519.PrivateKey
```

Comment:
Why do you keep the private key as ed25519 while we need x25519. I think it's better to convert the key before storing it on the NodeConfig from the start instead of changing it on start